syntax = 'proto3';
package api;
option go_package = "github.com/armadaproject/armada/pkg/api";

import "github.com/gogo/protobuf/gogoproto/gogo.proto";
import "pkg/api/submit.proto";
option (gogoproto.goproto_stringer_all) = false;
option (gogoproto.stringer_all) = true;


message JobRequest {
  string job_id = 1;
}

message JobResponse{
  Job job = 1;
}

message JobStatusRequest{
  string job_id = 1;
}

message JobStatusResponse{
  string job_id = 1;
  JobState job_state = 2;
}

service Jobs {
  rpc GetJobStatus (JobStatusRequest) returns (JobStatusResponse);
  rpc GetJob (JobRequest) returns (JobResponse);
}
